rcl/include/rcl/timer.h

```diff
@@ -41,6 +42,14 @@ typedef struct rcl_timer_s
   rcl_timer_impl_t * impl;
 } rcl_timer_t;
 
+/// Structure which encapsulates the on reset callback data
+typedef struct rcl_timer_on_reset_callback_data_s
+{
+  rcl_event_callback_t on_reset_callback;
+  const void * user_data;
+  size_t reset_counter;
```

Comment:
just a question. this needs to be `size_t`? why cannot be `bool`? I do not see how application makes difference, could you elaborate the actual use case?

Reply:
I would have used just a `bool` for this. But all the other entities callbacks have this `size_t` counter, so just in case I left it here.
If I have to be creative and find a use case, could be, I want to know how many times a timer had a backward time jump. If it has happened N times, I'd have this callback be called with N times as argument.

Reply:
@mauropasse thanks for the comments.